test/test_data/test_feature_validator.py

```diff
@@ -0,0 +1,355 @@
+    if the user changes the type (pandas then)
+    """
+    validator = FeatureValidator()
+
+    df = pd.DataFrame([
+        {'int': 1, 'float': 1.0, 'category': 'one', 'bool': True},
+        {'int': 2, 'float': 2.0, 'category': 'two', 'bool': False},
+    ])
+
+    for col in df.columns:
+        df[col] = df[col].astype(col)
+
+    enc_columns, feature_types = validator._get_columns_to_encode(df)
+
+    assert enc_columns == ['category', 'bool']
+    assert feature_types == ['numerical', 'numerical', 'categorical', 'categorical']
+
+
+def test_dataframe_input_unsupported():
+    """
+    Makes sure we raise a proper message to the user,
+    when providing not supported data input
+    """
+    validator = FeatureValidator()
+    with pytest.raises(ValueError, match=r"Auto-sklearn does not support time"):
+        validator.fit(
+            pd.DataFrame({'datetime': [pd.Timestamp('20180310')]})
+        )
+    with pytest.raises(ValueError, match="has invalid type object"):
+        validator.fit(
+            pd.DataFrame({'string': ['foo']})
+        )
+    with pytest.raises(ValueError, match=r"Auto-sklearn only supports.*yet, the provided input"):
+        validator.fit({'input1': 1, 'input2': 2})
+    with pytest.raises(ValueError, match=r"has unsupported dtype string"):
+        validator.fit(pd.DataFrame([{'A': 1, 'B': 2}], dtype='string'))
+
+
+@pytest.mark.parametrize(
+    'input_data_featuretest',
+    (
+        'numpy_numericalonly_nonan',
+        'numpy_numericalonly_nan',
+        'pandas_numericalonly_nonan',
+        'pandas_numericalonly_nan',
+        'list_numericalonly_nonan',
+        'list_numericalonly_nan',
+        # Category in numpy is handled via feat_type
+        'numpy_categoricalonly_nonan',
+        'numpy_mixed_nonan',
+        'numpy_categoricalonly_nan',
+        'numpy_mixed_nan',
+        'sparse',
+    ),
+    indirect=True
+)
+def test_no_encoder_created(input_data_featuretest):
+    """
+    Makes sure that for numerical only features, no encoder is created
+    """
+    validator = FeatureValidator()
+    validator.fit(input_data_featuretest)
+    validator.transform(input_data_featuretest)
+    assert validator.encoder is None
+
+
+@pytest.mark.parametrize(
+    'input_data_featuretest',
+    (
+        'pandas_categoricalonly_nonan',
+        'pandas_mixed_nonan',
+    ),
+    indirect=True
+)
+def test_encoder_created(input_data_featuretest):
+    """
+    This test ensures an encoder is created if categorical data is provided
+    """
+    validator = FeatureValidator()
+    validator.fit(input_data_featuretest)
+    transformed_X = validator.transform(input_data_featuretest)
+    assert validator.encoder is not None
+
+    # Make sure that the encoded features are actually encoded. Categorical columns are at
+    # the start after transformation. In our fixtures, this is also honored prior encode
+    enc_columns, feature_types = validator._get_columns_to_encode(input_data_featuretest)
```

Comment:
Would it make sense to test that there's at least one numerical and one categorical in `feature_types`?

Reply:
Done!